dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -96,9 +97,13 @@ public boolean isValidWorkflow(String content) {
 
     @Override
     public Map<String, SourceFile> processImports(String content, Version version, SourceCodeRepoInterface sourceCodeRepoInterface) {
+        return processImports(content, version, sourceCodeRepoInterface, new ArrayList<>());
+    }
+
+    public Map<String, SourceFile> processImports(String content, Version version, SourceCodeRepoInterface sourceCodeRepoInterface, List<String> alreadyImported) {
```

Comment:
I think this can be private now. 
It would be more efficient to use a set for `alreadyImported` than a List. 
Or to take it a step further, you can just pass along imports.